frontend/packages/dev-console/src/components/pipelines-lists/PipelinesListsPage.tsx

```diff
@@ -0,0 +1,116 @@
+import * as React from 'react';
+import { match as Rmatch } from 'react-router-dom';
+import MultiTabListPage from '../multi-tab-list/MultiTabListPage';
+import { referenceForModel, K8sKind } from '@console/internal/module/k8s';
+import {
+  PipelineModel,
+  PipelineResourceModel,
+  ConditionModel,
+  PipelineRunModel,
+} from '../../models';
+import { Page, Firehose } from '@console/internal/components/utils';
+import { filters } from '../pipelines/list-page/PipelineAugmentRuns';
+import PipelineAugmentRunsWrapper from '../pipelines/list-page/PipelineAugmentRunsWrapper';
+import { TechPreviewBadge } from '@console/shared';
+import PipelineRunsResourceList from '../pipelineruns/PipelineRunsResourceList';
+import { DefaultPage } from '@console/internal/components/default-resource';
+import PipelineResourcesListPage from '../pipeline-resources/list-page/PipelineResourcesListPage';
+
+interface PipelinesListPageProps {
+  match: Rmatch<any>;
+}
+
+export const PipelinesResourceList: React.FC<PipelinesListPageProps> = ({
+  match: {
+    params: { ns: namespace },
+  },
+}) => {
+  const resources = [
+    {
+      isList: true,
+      kind: referenceForModel(PipelineModel),
+      namespace,
+      prop: PipelineModel.id,
+      filters: { ...filters },
+    },
+  ];
+  return (
+    <div className="co-m-pane__body">
+      <Firehose resources={resources}>
+        <PipelineAugmentRunsWrapper />
+      </Firehose>
+    </div>
+  );
+};
+
+const PipelinesListPage: React.FC<PipelinesListPageProps> = ({ match }) => {
+  const {
+    params: { ns: namespace },
+  } = match;
+  const showTitle = false;
+  const showBadge = false;
+  const title = (
+    <span style={{ display: 'flex', alignItems: 'flex-end' }}>
+      Pipelines
+      <span style={{ marginLeft: 'var(--pf-global--spacer--md)' }}>
+        <TechPreviewBadge />
+      </span>
+    </span>
+  );
+  const createAction = (model: K8sKind, activeNamespace?: string) => {
+    let action = {
+      label: model.label,
+      href: activeNamespace
+        ? `/k8s/ns/${activeNamespace}/${referenceForModel(model)}/~new`
+        : `/k8s/cluster/${referenceForModel(model)}/~new`,
+    };
+    if (model === PipelineModel) {
+      const href = activeNamespace ? `${action.href}/builder` : action.href;
+      action = {
+        ...action,
+        href,
+      };
+    }
+    return action;
+  };
+  const menuActions = [
+    createAction(PipelineModel, namespace),
+    createAction(PipelineRunModel, namespace),
+    createAction(PipelineResourceModel, namespace),
+    createAction(ConditionModel, namespace),
+  ];
```

Comment:
So in this PR my plan was to give an end to end flow so that the `MultiTabListPage` comp can be tested properly. The code to create the actions is generic and can be used for other pages that I will create in the next story, so I am moving it out of this comp and adding in the utils file.

> You're not handling the "all-projects" case that Admin has. If I go to all-projects and create a Pipeline, I get into a "cluster" Pipeline YAML page... not the builder.

I didnt get this though, even currently when `all-projects` is selected the cluster url is used, if you see the `PipelinesResourceList.tsx`

Reply:
> So in this PR my plan was to give an end to end flow so that the MultiTabListPage comp can be tested properly. The code to create the actions is generic and can be used for other pages...

Other pages as in non `MultiTabListPage` usages? This has some value for sure. Not sure the best way to go about placing that logic somewhere useful though. This code sits in dev-console today, I think we'd need to find a better spot for it if we wanted to think more generically. Today, I think it's okay to keep it in dev-console before we throw it into another package.

The main takeaway here is the `MultiTabListPage` component should handle listing the tabs (which show another component), a title, an optional badge and a dropdown for actions. That way anyone using this component gets the full page experience out of the box.

>> You're not handling the "all-projects" case that Admin has. If I go to all-projects and create a Pipeline, I get into a "cluster" Pipeline YAML page... not the builder.
>
> I didnt get this though, even currently when all-projects is selected the cluster url is used, if you see the PipelinesResourceList.tsx

You shouldn't be going to the `.../cluster/...` page for any of the resources that are namespaced based. Pipeline Builder is not reached with the current implemented logic and all the other resources are supposed to be in a namespace. Today the create buttons for other lists assume `default` namespace when in `all-projects`... so I went with that logic in my snippet above. Does that make sense? We can talk more in slack if needed.

Reply:
> You shouldn't be going to the .../cluster/... page for any of the resources that are namespaced based. Pipeline Builder is not reached with the current implemented logic and all the other resources are supposed to be in a namespace. Today the create buttons for other lists assume default namespace when in all-projects... so I went with that logic in my snippet above. Does that make sense? We can talk more in slack if needed.

Yeah I can do that here as well but as I told above in PipelineResourceList.tsx it redirects to cluster URL, that is why I got confused.
```
  createProps={{
        to: namespace
          ? `/k8s/ns/${namespace}/${referenceForModel(PipelineModel)}/~new/builder`
          : `/k8s/cluster/${referenceForModel(PipelineModel)}/~new`,
      }}
```

Reply:
I understand that, that's point 2. Of course you can fix it here... not sure why you don't want to do it as part of the component that renders the links. It's K8sKind based logic you want to do, you can just generically handle it as part of this component. 

What's the reasoning behind not wanting to add it to `MultiTabListPage`?

Reply:
I was actually talking about using the default ns when all-projects is selected. No issue with adding it in the tab component 😛